src/engine/source/api/src/test/handlers.cpp

```diff
@@ -0,0 +1,412 @@
+{
+
+namespace eEngine = ::com::wazuh::api::engine;
+namespace eTest = ::com::wazuh::api::engine::test;
+
+/**
+ * @brief Get the minimum available priority for a route.
+ *
+ * @param router Router instance.
+ * @return int32_t Minimum available priority. If no priority is available, returns -1.
+ */
+static inline int32_t getMinimumAvailablePriority(const shared_ptr<::router::Router>& router)
+{
+    // Create a set to store the taken priorities given the table
+    std::unordered_set<uint32_t> takenPriorities;
+
+    const auto routerTable = router->getRouteTable();
+    for (const auto& [name, priority, filter, policy] : routerTable)
+    {
+        takenPriorities.insert(priority);
+    }
+
+    int32_t minAvailablePriority {MINIMUM_PRIORITY};
+
+    // The condition may be confusing as the actual priority increases while the value decreases
+    while (takenPriorities.count(minAvailablePriority) > 0 && MAXIMUM_PRIORITY > minAvailablePriority)
+    {
+        // If priority is taken, decrease the value (so, increase the priority)
+        minAvailablePriority--;
+    }
+
+    return minAvailablePriority;
+}
+
+/**
+ * @brief Add a filter to the catalog.
+ *
+ * @param catalog Catalog instance.
+ * @param filterName Filter name.
+ * @param filterContent Filter content.
+ * @return std::optional<base::Error> If an error occurs, returns the error. Otherwise, returns std::nullopt.
+ */
+static inline std::optional<base::Error>
+addFilterToCatalog(shared_ptr<catalog::Catalog> catalog, const string& filterName, const string& filterContent)
+{
+    std::string error;
+
+    catalog::Resource targetResource;
+
+    try
+    {
+        targetResource = catalog::Resource {base::Name {"filter"}, catalog::Resource::Format::json};
+    }
+    catch (const std::exception& e)
+    {
+        error = e.what();
+    }
+
+    // If no error occurred, post the resource
+    if (error.empty())
+    {
+        const auto postResourceError = catalog->postResource(targetResource, filterContent);
+        if (postResourceError)
+        {
+            error = postResourceError.value().message;
+        }
+    }
+
+    return (error.empty() ? std::nullopt : std::make_optional(base::Error {error}));
+}
+
+/**
+ * @brief Delete a filter from the catalog.
+ *
+ * @param filterName Filter name.
+ * @param catalog Catalog instance.
+ * @return std::optional<base::Error> If an error occurs, returns the error. Otherwise, returns std::nullopt.
+ */
+static inline std::optional<base::Error> deleteFilterFromStore(const string& filterName,
+                                                               shared_ptr<catalog::Catalog> catalog)
+{
+    std::string error;
+
+    // Build target resource
+    catalog::Resource targetResource;
+    try
+    {
+        base::Name name {filterName};
+        targetResource = catalog::Resource {name, catalog::Resource::Format::json};
+    }
+    catch (const std::exception& e)
+    {
+        error = e.what();
+    }
```

Comment:
**thought:** catalog::Resource constructor only throws std::runtime_error and here we are catching std::exception. Should we add a default catch(...) block here or add the **noexcept** modifier to `deleteFilterFromStore` . It seems that the idea is to wrap any exception to the base::Error message. Please disregard if any other code in the stack deals with any other exception, in which case no re-trhown is here to bubble them up.